openhcl/virt_mshv_vtl/src/processor/snp/mod.rs

```diff
@@ -376,7 +376,11 @@ impl BackingPrivate for SnpBacked {
         unreachable!("extint managed through software apic")
     }
 
-    fn request_untrusted_sint_readiness(this: &mut UhProcessor<'_, Self>, sints: u16) {
+    fn request_untrusted_sint_readiness(
+        this: &mut UhProcessor<'_, Self>,
+        vtl: GuestVtl,
+        sints: u16,
+    ) {
         if this.backing.hv_sint_notifications & !sints == 0 {
```

Comment:
Do we need to have two hv_sint_notifications, one per vtl?

Reply:
At a glance, it seems like we could end up in `request_untrusted_sint_readiness` for either VTL, so I would say yes. But `hv_sint_notifications` is only cleared in `handle_synic_deliverable_exit`, which only operates on VTL0.

@sluck-msft Is it possible to get to `request_untrusted_sint_readiness` for VTL1?

Reply:
I defer to Steven as the expert on apic and interrupt state, but I think at the moment we don't expect to get untrusted sints for VTL 1. It is theoretically possible though, if we want to future-proof.

Reply:
I am far from an expert haha. But if we're only expecting and handling these for VTL 0 then we either should assert on the param, or just not take it at all.

Reply:
Let me add an assert for now, and then we can relax it once we handle VTL1 here.